app/javascript/components/organisms/StatusHistoryPopover.js

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from 'react'
+import classNames from 'classnames'
+import PropTypes from 'prop-types'
+
+import Icon from '~/components/atoms/Icon'
+import Popover from '~/components/molecules/Popover'
+import StatusItems from '~/components/molecules/lease_up_sidebar/StatusItems'
+import Loading from '../molecules/Loading'
+import apiService from '~/apiService'
+
+const MAX_UPDATES_TO_SHOW_DEFAULT = 4
+const StatusHistoryPopover = ({ applicationId }) => {
+  const [showingAllStatuses, setShowingAllStatuses] = useState(false)
+  const onShowHideStatusesToggled = () => setShowingAllStatuses(!showingAllStatuses)
+
+  const StatusIconButton = ({ onClick, ref }) => (
+    <button ref={ref} onClick={onClick} className='tiny icon margin-left--half'>
+      <Icon icon='list-unordered' size='medium' />
+    </button>
+  )
+  const [data, setData] = useState({ statusItems: [], loading: false })
+
+  const fetchStatusHistory = async () => {
+    try {
+      setData({ statusItems: data.statusItems, loading: true })
+      const items = await apiService.getFieldUpdateComments(applicationId)
+      setData({ statusItems: items, loading: false })
+    } catch (e) {
+      setData({ statusItems: data.statusItems, loading: false })
+    }
+  }
+
+  const onClick = async () => {
+    data.statusItems.length === 0 && (await fetchStatusHistory())
+  }
+
+  const numberOfStatusesToDisplay = showingAllStatuses
+    ? data.statusItems.length
+    : MAX_UPDATES_TO_SHOW_DEFAULT
+
+  return (
+    <Popover buttonElement={StatusIconButton} onButtonClick={onClick}>
+      {data.loading ? (
+        <Loading isLoading />
+      ) : (
+        <>
+          <StatusItems
+            statusItems={data.statusItems}
+            limit={numberOfStatusesToDisplay}
+            height='20rem'
```

Comment:
I'm a little worried about styles getting hidden in render functions. How do you feel about pulling all styling into constants at the top of the file it's used in? That way it's all in one place and would be easy to convert to styledComponents or whatever if we move to something like that.

So this would just be:
```
...imports...

// could also be a function if we're calculating style from props or state
const StatusItemStyle = {
  height: '20rem'
}

...

<StatusItems style={StatusItemStyle}>
```

Reply:
Edit: actually I just realized `height` is a specific prop on StatusItems, so ignore this comment in this case, but are you cool with that pattern for other times we use inline styles?

Reply:
yep I think that's reasonable